Assignment4/buttaciemanuel-230777/task07.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1yPniC8cCIRK0ofpUQK_h0bcNOa4CroCF
+
+**Task 07: Querying RDF(s)**
+"""
+
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from queue import Queue
+from rdflib import Graph, Namespace, Literal, XSD
+from rdflib.namespace import RDF, RDFS, FOAF
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0/")
+
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# 1. rdflib
+subclasses = Queue()
+visited = set()
+
+for s, _, _ in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    subclasses.put(s)
+    visited.add(s)
+
+while not subclasses.empty():
+    next = subclasses.get()
+    
+    for s, _, _ in g.triples((None, RDFS.subClassOf, next)):
+        subclasses.put(s)
+
+    visited.add(next)
+
+for subclass in visited:
+    print(subclass)
```

Comment:
This is not correct, as it does not print all the subclass hierarchy. Use a recursive function